### PR TITLE
Sort plans alphabetically in Create Application form

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

Because it's required

**Which issue(s) this PR fixes** 

Bug reported for [THREESCALE-6879: [3scale][2.11][HI-prio] Add 'Create new Application' flow to Product > Applications index](https://issues.redhat.com/browse/THREESCALE-6879)

**Note**: plans in the images have been created in reverse order so that it doesn't coincide with created_at order.
Screenshot 2021-11-16 at 13 10 34
Screenshot 2021-11-16 at 13 10 28

### CODE REVIEW

**Review by akostadinov — COMMENTED** [review]
Looks nice. Do we have pagination to care about here or not of a concern?

Also one wording trivial comment.

File: test/unit/presenters/service_presenter_test.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ServicePresenterTest < ActiveSupport::TestCase
+  test 'new application application and service plans are sorted alphabetically' do
```
[the remark this answers is not in the document]
Comment (hallelujah) [reply test/unit/presenters/service_presenter_test.rb:6]: I guess it is `application plan`

```suggestion
  test 'new application plan and service plans are sorted alphabetically' do
```

**Comment by josemigallas** [comment]
> Looks nice. Do we have pagination to care about here or not of a concern?
> 
> Also one wording trivial comment.

No pagination in this collection

**Review by akostadinov — DISMISSED** [review]

File: test/unit/presenters/service_presenter_test.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    FactoryBot.create(:service_plan, issuer: service, position: 3, name: 'Service Plan 1')
+    FactoryBot.create(:service_plan, issuer: service, position: 2, name: 'Service Plan 2')
+
+    data = ServicePresenter.new(service).new_application_data
+
+    app_plans = data[:appPlans]
+    assert_equal app_plans, (app_plans.sort_by { |p| p[:name] })
+
+    service_plans = data[:servicePlans]
+    assert_equal service_plans, (service_plans.sort_by { |p| p[:name] })
```
Comment (akostadinov) [inline test/unit/presenters/service_presenter_test.rb:22]: huh, a nitpick, here the expected value seems to be `(service_plans.sort_by { |p| p[:name] })` and actual value `service_plans`.
```suggestion
    assert_equal (service_plans.sort_by { |p| p[:name] }), service_plans
```

File: test/unit/presenters/service_presenter_test.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    FactoryBot.create(:application_plan, issuer: service, position: 2, name: 'Application Plan 2')
+
+    FactoryBot.create(:service_plan, issuer: service, position: 1, name: 'Service Plan 3')
+    FactoryBot.create(:service_plan, issuer: service, position: 3, name: 'Service Plan 1')
+    FactoryBot.create(:service_plan, issuer: service, position: 2, name: 'Service Plan 2')
+
+    data = ServicePresenter.new(service).new_application_data
+
+    app_plans = data[:appPlans]
+    assert_equal app_plans, (app_plans.sort_by { |p| p[:name] })
```
Comment (akostadinov) [inline test/unit/presenters/service_presenter_test.rb:19]:
```suggestion
    assert_equal (app_plans.sort_by { |p| p[:name] }), app_plans
```

**Comment by josemigallas** [comment]
@akostadinov

**Review by akostadinov — APPROVED** [review]

**Pull request merged**